test/extensions/filters/listener/tls_inspector/tls_inspector_test.cc

```diff
@@ -232,6 +237,166 @@ TEST_P(TlsInspectorTest, ClientHelloTooBig) {
   EXPECT_EQ(1, cfg_->stats().client_hello_too_large_.value());
 }
 
+// Test that the filter sets the ja3 hash
+TEST_P(TlsInspectorTest, ConnectionFingerprint) {
+  envoy::extensions::filters::listener::tls_inspector::v3::TlsInspector proto_config;
+  proto_config.set_enable_ja3_fingerprinting(true);
+  cfg_ = std::make_shared<Config>(store_, proto_config);
+  std::vector<uint8_t> client_hello =
+      Tls::Test::generateClientHello(std::get<0>(GetParam()), std::get<1>(GetParam()), "", "");
+  init();
+  EXPECT_CALL(os_sys_calls_, recv(42, _, _, MSG_PEEK))
+      .WillOnce(Invoke(
+          [&client_hello](os_fd_t, void* buffer, size_t length, int) -> Api::SysCallSizeResult {
+            ASSERT(length >= client_hello.size());
+            memcpy(buffer, client_hello.data(), client_hello.size());
+            return Api::SysCallSizeResult{ssize_t(client_hello.size()), 0};
+          }));
+  EXPECT_CALL(socket_, setJA3Hash(_));
+  EXPECT_CALL(socket_, setRequestedServerName(_)).Times(0);
+  EXPECT_CALL(socket_, setRequestedApplicationProtocols(_)).Times(0);
+  EXPECT_CALL(socket_, setDetectedTransportProtocol(absl::string_view("tls")));
+  EXPECT_CALL(cb_, continueFilterChain(true));
+  file_event_callback_(Event::FileReadyType::Read);
+}
+
+// Test that the filter sets the correct ja3 hash.
+// Fingerprint created with User-Agent "curl/7.64.1" and a request to ja3er.com/json.
+TEST_P(TlsInspectorTest, ConnectionJA3Hash) {
+  envoy::extensions::filters::listener::tls_inspector::v3::TlsInspector proto_config;
+  proto_config.set_enable_ja3_fingerprinting(true);
+  cfg_ = std::make_shared<Config>(store_, proto_config);
+  std::vector<uint8_t> client_hello = Tls::Test::generateClientHelloFromJA3Fingerprint(
```

Comment:
Make a helper function used by many of these test cases that takes an input of the ja3 fingerprint, and it does all the test actions and expectations, so that most of these test cases become only a couple lines.